enterprise/tailnet/handshaker_test.go

```diff
@@ -0,0 +1,69 @@
+package tailnet_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"tailscale.com/types/key"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/slogtest"
+	"github.com/coder/coder/v2/coderd/database/dbtestutil"
+	"github.com/coder/coder/v2/enterprise/tailnet"
+	agpl "github.com/coder/coder/v2/tailnet"
+	"github.com/coder/coder/v2/tailnet/proto"
+	"github.com/coder/coder/v2/testutil"
+)
+
+func TestPGCoordinatorDual_ReadyForHandshake_OK(t *testing.T) {
+	t.Parallel()
+	if !dbtestutil.WillUsePostgres() {
+		t.Skip("test only with postgres")
+	}
+	store, ps := dbtestutil.NewDB(t)
+	ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitSuperLong)
+	defer cancel()
+	logger := slogtest.Make(t, nil).Leveled(slog.LevelDebug)
+	coord1, err := tailnet.NewPGCoord(ctx, logger.Named("coord1"), ps, store)
+	require.NoError(t, err)
+	defer coord1.Close()
+	coord2, err := tailnet.NewPGCoord(ctx, logger.Named("coord2"), ps, store)
+	require.NoError(t, err)
+	defer coord2.Close()
+
+	clientID, agentID := uuid.New(), uuid.New()
+
+	cReq, cRes := coord1.Coordinate(ctx, clientID, clientID.String(), agpl.ClientCoordinateeAuth{AgentID: agentID})
+	aReq, aRes := coord2.Coordinate(ctx, agentID, agentID.String(), agpl.AgentCoordinateeAuth{ID: agentID})
```

Comment:
this test will be fewer lines and more readable if you use `tailnet/test/peer.go` framework.  Just need to extend it to include ReadyForHandshake.